### PR TITLE
feat(profile): wire up profile to warp

### PR DESCRIPTION
**Pull request by sdwoodbury**
### What this PR does 📖

- allow user to update Identity: status, profile picture, banner 
- identities for friends are refreshed every minute
- the profile picture and status appear in the top bar 
- the pfp in the sidebar isn't implemented yet, and should probably be held off on until we decide if we will change the data model in `state` per discussions on the Uplink private discord channel. 
- same goes for other places that the profile picture and online status need to be wired up

### Which issue(s) this PR fixes 🔨

- Resolve #

### Additional comments 🎤
- the input box for user status remains green after "enter" is pressed. this eventually needs to be changed. 
- the user's Identity is now fetched from Warp (minor change). seems the other stuff - whether the user is online and their platform - can be determined at runtime and doesn't need to be saved.

### CODE REVIEW

**Comment by dariusc93** [comment]
~~Note: This should not merge until #375 has been merged into dev~~

**Review by phillsatellite — DISMISSED** [review]

**Comment by phillsatellite** [comment]
Approved on my side, I did have some issues here and there where a friend would go missing and not appear for about 3 mins then it seemed like Warp finally loaded the User info. I think this is more of a network issue on my side because I tested on dev and had the same issues. When Warp was running really good then every function on this PR would work really well. I can create a ticket for the wiring pfp to the sidebar and other needed places if thats cool with you @sdwoodbury was waiting for the conclusion for the state discussion  I did mention to Stu that I have ticket #362 that I will be adding more info about the green indicator around the input boxes.

**Comment by dariusc93** [comment]
> Approved on my side, I did have some issues here and there where a friend would go missing and not appear for about 3 mins then it seemed like Warp finally loaded the User info. I think this is more of a network issue on my side because I tested on dev and had the same issues. When Warp was running really good then every function on this PR would work really well. I can create a ticket for the wiring pfp to the sidebar and other needed places if thats cool with you @sdwoodbury was waiting for the conclusion for the state discussion I did mention to Stu that I have ticket #362 that I will be adding more info about the green indicator around the input boxes.

Hard to really say if its an issue with warp or if its related to uplink. How often does this happen? Is it when you send, receive, or after accept a request that it disappears or not show up? Does anything appear in the logs?

File: common/src/state/identity.rs
Comment (dariusc93) [inline common/src/state/identity.rs:76]: `Identity::from` (and to that extent, `Identity::into`) could be used since it would being `multipass::identity::Identity` into this wrapper.

File: common/src/warp_runner/manager/commands/multipass_commands.rs
Comment (dariusc93) [inline common/src/warp_runner/manager/commands/multipass_commands.rs]: Could `Identifier::did_keys` be better here? That way you could pass `ids` into the identifier when calling `MultiPass::get_identity` to get the list itself then construct the hashmap from that (or just the list of `Identity`?)

eg something like this should suffice 

```rust
async fn multipass_refresh_friends(
    account: &mut Account,
) -> Result<HashMap<DID, multipass::identity::Identity>, Error> {
    let ids = account.list_friends().await?;

    let list = account.get_identity(Identifier::did_keys(ids.clone())).await?;
    let friends: HashMap<DID, identity::Identity> = list.iter().map(|ident| (ident.did_key(), ident.clone())).collect();
    if list.is_empty() {
        log::warn!("No identities found");
    }
    Ok(friends)
}

```

Comment (dariusc93) [reply common/src/warp_runner/manager/commands/multipass_commands.rs]: Note: Could also compare the `ids` against the map and log any identities not found too.

**Review by dariusc93 — DISMISSED** [review]
LGTM. Left some comments and suggestions but non-blocking to this PR

File: common/src/warp_runner/manager/commands/multipass_commands.rs
Comment (sdwoodbury) [reply common/src/warp_runner/manager/commands/multipass_commands.rs]: I like this suggestion. I didn't realize I could do this. That's a nice feature of MultiPass.

**Review by tooshel — APPROVED** [review]

**Pull request merged**
